test/e2e/text.cy.js

```diff
@@ -0,0 +1,22 @@
+describe('Text', function () {
+  beforeEach(function () {
+    cy.getData().then(function (data) {
+      this.data = data;
+      cy.visit('/');
+    });
```

Comment:
This does the same thing:
```suggestion
    cy.getData();
    cy.visit('/');
```
This isn't the await/async promise syntax, it's a bespoke cypress promise-like syntax for generating some execution order. `cy.getData()`, without the `then` subclause completes before `cy.visit('/')` I believe.

`cy.getData()` does `this.data = data` already at https://github.com/adaptlearning/adapt_framework/blob/ad3913fb8a173b5a2d06c1dd7df0f91a6ab0881e/test/e2e/commands.js#L29

Reply:
You can navigate to a component in adapt by routing to `#/id/c-15` etc

Reply:
`this.data` in each test has the shortcuts to subsets of the main array `this.data = [{ _id: "", ... }, ...]`:
`this.data.contentObjects, this.data.articles, this.data.blocks, this.data.components, ` you don't need to recreate these with filters. https://github.com/adaptlearning/adapt_framework/blob/ad3913fb8a173b5a2d06c1dd7df0f91a6ab0881e/test/e2e/commands.js#L32-L63